src/Build/Definition/Project.cs

```diff
@@ -1117,33 +1129,169 @@ public List<GlobResult> GetAllGlobs(string itemType)
             return GetAllGlobs(GetItemElementsByType(GetEvaluatedItemElements(), itemType));
         }
 
+        // represents cumulated remove information for a particular item type
+        private struct CumulativeRemoveElementData
+        {
+            private ImmutableList<IMSBuildGlob>.Builder _globs;
+            private ImmutableHashSet<string>.Builder _fragmentStrings;
+
+            public IEnumerable<IMSBuildGlob> Globs => _globs.ToImmutable();
```

Comment:
If you're returning an `IEnumerable` is there any point in making it immutable? It's already readonly.